common/persistence/versionhistory/version_history.go

```diff
@@ -164,23 +161,23 @@ func IsLCAVersionHistoryItemAppendable(v *historyspb.VersionHistory, lcaItem *hi
 		panic("lcaItem is nil")
 	}
 
-	return *v.Items[len(v.Items)-1] == *lcaItem
+	return IsEqualVersionHistoryItem(v.Items[len(v.Items)-1], lcaItem)
```

Comment:
I don't understand what is the difference. Don't you trust go in struct comparisons?